README.md

```diff
@@ -1,172 +1,235 @@
+programs.
 
-###### ColorSafe https://github.com/colorsafe/colorsafe
+### [ColorSafe](https://github.com/colorsafe/colorsafe)
 
-A data matrix for printing on paper. Inspired by PaperBak, ColorSafe is written with modern methods and technologies and is cross-platform. It aims to allow a few Megabytes of data (or more) to be stored on paper for a worst case scenario backup, for extremely long-term archiving, or just for fun. With best practices, ColorSafe encoded data can safely withstand the viccissitudes of technology changes over long periods of time.
+A data matrix for printing on paper. Inspired by PaperBak, ColorSafe is
+written with modern methods and technologies and is cross-platform. It
+aims to allow a few megabytes of data (or more) to be stored on paper
+for a worst case scenario backup, for extremely long-term archiving, or
+just for fun. With best practices, ColorSafe encoded data can safely
+withstand the viccissitudes of technology changes over long periods of
+time.
 
-###### Twibright Optar http://ronja.twibright.com/optar/
+### [Twibright Optar](http://ronja.twibright.com/optar/)
 
-Uses the not-so-common Golay code to backup 200KB per page. So it offers a much higher
-density than paperbackup.py, but is probably more affected by defects on the paper.
-GPL 2 and designed for Linux.
+Uses the not-so-common Golay code to backup 200KB per page. So it offers
+a much higher density than paperbackup.py, but is probably more affected
+by defects on the paper. It is licensed under GPLv2 and designed for
+Linux.
 
-###### Paperkey http://www.jabberwocky.com/software/paperkey/
+### [Paperkey](http://www.jabberwocky.com/software/paperkey/)
 
-It is designed to reduce the data needed to backup a private GnuPG key. It does not help you
-to print and scan the data. So it could be used in addition to paperbackup.py.
+It is designed to reduce the data needed to backup a private GnuPG key.
+It does not help you to print and scan the data. So it could be used in
+addition to paperbackup.py.
 
-###### asc2qr.sh https://github.com/4bitfocus/asc-key-to-qr-code
+### [asc2qr.sh](https://github.com/4bitfocus/asc-key-to-qr-code)
 
-Very similar to paperbackup.py. But it only outputs .png images without ordering information.
-So you have to arrange printing and ordering yourself.
+Very similar to paperbackup.py. But it only outputs PNG images without
+ordering information, so you have to arrange printing and ordering
+yourself.
 
 ## License
 
-MIT X11 License
+[MIT X11 License](https://opensource.org/licenses/MIT)
+<!-- SPDX-License-Identifier: MIT -->
+
+*NB: The upstream author marked this as "MIT X11" (there's an MIT
+license, and an X11 license, but there's no SPDX identifier for an "MIT
+X11" license) but included the MIT license in the
+[LICENSE](https://github.com/intra2net/paperbackup/blob/master/LICENSE)
+file. Please contact
+[upstream](https://github.com/intra2net/paperbackup) if you're confused
+about which license(s) apply.*
```

Comment:
i don't think upstream is ambiguous at all here actually. there's a LICENSE file right here:

https://github.com/intra2net/paperbackup/blob/master/LICENSE

sure, it says "x11" here, but that's probably an oversight.

it looks like a plain old expat to me, so maybe you'd like to clarify *that* way instead of introducing more ambiguity. :)

in other words:

```suggestion
[MIT "Expat" License](https://opensource.org/licenses/MIT)
<!-- SPDX-License-Identifier: MIT -->
```